### ISSUE TITLE
Update Async, FieldValue.Delete not working on multiple fields

### ISSUE THREAD

**Issue by npicouet** [firebase/quickstart-unity#882]
### [REQUIRED] Please fill in the following fields:

  * Unity editor version: 2019.4.12f1
  * Firebase Unity SDK version: 6.16.1
  * Source you installed the SDK: UPM
  * Problematic Firebase Component: Firestore
  * Other Firebase Components in use: Everything
  * Additional SDKs you are using: Facebook, Google Sign In, Apple Sign In
  * Platform you are using the Unity editor on: Mac
  * Platform you are targeting: Mobile (iOS and Android)
  * Scripting Runtime: IL2CPP

### [REQUIRED] Please describe the issue here:

When attempting to delete multiple fields in an UpdateAsync call, only the last field in the array actually gets deleted.

#### Steps to reproduce:

What's the issue repro rate? 100%

What happened? How can we make the problem occur?
Expected all the fields to be deleted. Only one gets deleted. Specifically in my case, only "Rooms" field gets deleted. If I comment out "Rooms", then "Pending Rooms" gets deleted. It's only ever 1 field at a time, even though I want to delete them all in my update.

#### Relevant Code:

```
//Create Dictionary
Dictionary<string, object> userUpdate = new Dictionary<string, object>()
{
	{ "Quickplay Rooms", FieldValue.Delete },
	{ "Pending Rooms", FieldValue.Delete },
	{ "Rooms", FieldValue.Delete },
	// { "Room Version", SaveManager.RoomVersion }
};

//Update User Profile
docRef.UpdateAsync(userUpdate).ContinueWithOnMainThread(innerTask =>
{
	//Callback code in here
});
```

**Comment by dconeybe** [comment]
Thank you for reporting this. I have been able to reproduce and will investigate.

**Comment by dconeybe** [comment]
Update: I've found the root cause and it is indeed a bug. In the special handling of `FieldValue.Delete`, a `break` statement is incorrectly being used instead of a `continue` statement: https://github.com/firebase/firebase-cpp-sdk/blob/f0a9d5b65e3a014aee347ea9bb3b664cdce2ae55/firestore/src/ios/user_data_converter_ios.cc#L481. 

With this bug present, including `FieldValue.Delete` in an update causes some of the fields being requested to be updated to be excluded from the update.

Unfortunately, I do not know if the fix will make it into the next release. I will update this thread once I can confirm. Fortunately, there are feasible workarounds.

#### Workaround 1: UpdateAsync() with exactly one key/value pair for delete

In a call to UpdateAsync(), either include *no* `FieldValue.Delete` values or the only key/value pair should be the delete. Since the first "delete" is processed correctly, but potentially causes other fields to be incorrectly excluded, having the *only* key being updated to a `FieldValue.Delete` value will behave correctly.

#### Workaround 2: Use WriteBatch

Fortunately, `WriteBatch` is not affected by this bug as long as the fields to be deleted are added to the write batch one at a time (instead of all at once in a single Dictionary). For example, the code in the opening comment could be re-written as follows:

```
docRef.Firestore.StartBatch()
    .Update(docRef, "Quickplay Rooms", FieldValue.Delete)
    .Update(docRef, "Pending Rooms", FieldValue.Delete)
    .Update(docRef, "Rooms", FieldValue.Delete)
    .CommitAsync().ContinueWithOnMainThread(innerTask =>
        {
	    //Callback code in here
        });
```

**Comment by npicouet** [comment]
Thank you for looking into it, and thank you for the workarounds!
Glad it was something that can be fixed.

**Comment by dconeybe** [comment]
Update: The fix for this bug has been submitted. It did not make it into the Unity SDK 7.0.0 release but should make it into the next release (for which I do not have an ETA). Until then, please use one of the workarounds documented in the earlier comment https://github.com/firebase/quickstart-unity/issues/882#issuecomment-737448853.

Since there are no further actions to take to address this issue I am going to close it. Feel free to re-open if the bug is not resolved in a future release with release notes entry that reads something like this:

> Fixed partial updates in UpdateAsync() with FieldValue.Delete

Note to Googlers: The internal ticket is b/174676322 and the fix is cl/345664484.

**Comment by dconeybe** [comment]
The fix for this bug is included in today's Unity SDK 7.1.0 release.
